src/Native/System.Private.CoreLib.Native/pal_exepath.cpp

```diff
@@ -0,0 +1,93 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include <cstdlib>
+#include <cstring>
+#include <assert.h>
+#include <dirent.h>
+#include <dlfcn.h>
+#include <limits.h>
+#include <set>
+#include <string>
+#include <string.h>
+#include <sys/stat.h>
+#if defined(__FreeBSD__)
+#include <sys/types.h>
+#include <sys/param.h>
+#endif
+#if defined(HAVE_SYS_SYSCTL_H) || defined(__FreeBSD__)
+#include <sys/sysctl.h>
+#endif
+#if defined(__APPLE__)
+#include <mach-o/dyld.h>
+#endif
+#include <unistd.h>
+
+#if defined(__linux__)
+#define symlinkEntrypointExecutable "/proc/self/exe"
+#elif !defined(__APPLE__)
+#define symlinkEntrypointExecutable "/proc/curproc/exe"
+#endif
+
+extern "C" bool CoreLibNative_GetEntrypointExecutableAbsolutePath(char **buf)
```

Comment:
This behavior makes sense once we have optional loading of IL ECMA assemblies in CoreRT. 

I am wondering the simulator of the probing path makes sense at all for the native binaries produced by the full AOT. Why can't we just set RPATH on the native binary? It is what how one would deal with this problem on regular native binaries on Unix.